### DISCUSSION TITLE
Add a file explorer to view?

### DISCUSSION THREAD

**Discussion by diggyk** [microsoft/vscode-discussions#346] [Extension Development QnA] [answered]
I'm wondering if I can make an extension that providers a view that has a file explorer built in?  I want to add a custom FileSystemProvider but I don't want it tied into the file explorer directly and want it to appear in a different panel.  Is that doable?

**Comment by r3m0t** [comment, marked as answer]
No there can only be one file explorer.

But you can write your own tree view provider and create a command to open documents on click of a file item. The document could be in a custom file system. You'll just miss some behaviours like renaming a file or creating a new file with a text box inside the tree.
